File: Domain/Enums/PermissaoSistemaEnum.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace AspNetCoreApiSample.Domain.Enums
{
    /// <summary>
    /// Enum de permissões do sistema
    /// </summary>
    public enum PermissaoSistemaEnum
    {
        UsuarioConsulta = 1,
        UsuarioCria = 2,
        UsuarioEdita = 3,
        UsuarioRemove = 4,
    }
}

File: Web/ClientApp/src/app/_enums/PermissaoSistemaEnum.ts
export enum PermissaoSistemaEnum {
  UsuarioConsulta = 1,
  UsuarioCria = 2,
  UsuarioEdita = 3,
  UsuarioRemove = 4
}

export const PermissaoSistemaEnumLabel = new Map<PermissaoSistemaEnum, string>([
  [PermissaoSistemaEnum.UsuarioConsulta, 'Consulta de Usuários'],
  [PermissaoSistemaEnum.UsuarioCria, 'Criação de Usuários'],
  [PermissaoSistemaEnum.UsuarioEdita, 'Edição de Usuários'],
  [PermissaoSistemaEnum.UsuarioRemove, 'Exclusão de Usuários'],
]);
